resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -176,16 +179,6 @@ private[spark] abstract class YarnSchedulerBackend(
   }
 
   /**
-   * Reset the state of SchedulerBackend to the initial state. This is happened when AM is failed
-   * and re-registered itself to driver after a failure. The stale state in driver should be
-   * cleaned.
-   */
-  override protected def reset(): Unit = {
```

Comment:
I think `ExecutorAllocationManager#reset` is still necessary, but the following code should be removed
```scala
    initializing = true
    numExecutorsTarget = initialNumExecutors
    numExecutorsToAdd = 1
```